pkg/sdk/database_role_dto_builders.go

```diff
@@ -0,0 +1,110 @@
+package sdk
+
+func NewCreateDatabaseRoleRequest(
+	name DatabaseObjectIdentifier,
+) *CreateDatabaseRoleRequest {
+	s := CreateDatabaseRoleRequest{}
+	s.name = name
+	return &s
+}
+
+func (s *CreateDatabaseRoleRequest) WithOrReplace(orReplace bool) *CreateDatabaseRoleRequest {
+	s.orReplace = orReplace
+	return s
+}
+
+func (s *CreateDatabaseRoleRequest) WithIfNotExists(ifNotExists bool) *CreateDatabaseRoleRequest {
+	s.ifNotExists = ifNotExists
+	return s
+}
+
+func (s *CreateDatabaseRoleRequest) WithComment(comment *string) *CreateDatabaseRoleRequest {
+	s.comment = comment
+	return s
+}
+
+func NewAlterDatabaseRoleRequest(
+	name DatabaseObjectIdentifier,
+) *AlterDatabaseRoleRequest {
+	s := AlterDatabaseRoleRequest{}
+	s.name = name
+	return &s
+}
+
+func (s *AlterDatabaseRoleRequest) WithIfExists(ifExists bool) *AlterDatabaseRoleRequest {
+	s.ifExists = ifExists
+	return s
+}
+
+func (s *AlterDatabaseRoleRequest) WithRename(rename *DatabaseRoleRenameRequest) *AlterDatabaseRoleRequest {
+	s.clearForOneOf()
+	s.rename = rename
+	return s
+}
+
+func (s *AlterDatabaseRoleRequest) WithSet(set *DatabaseRoleSetRequest) *AlterDatabaseRoleRequest {
+	s.clearForOneOf()
+	s.set = set
+	return s
+}
+
+func (s *AlterDatabaseRoleRequest) WithUnsetComment() *AlterDatabaseRoleRequest {
+	s.clearForOneOf()
+	s.unset = NewDatabaseRoleUnsetRequest()
+	return s
+}
+
+func (s *AlterDatabaseRoleRequest) clearForOneOf() *AlterDatabaseRoleRequest {
```

Comment:
That's certainly an option but I wonder if just returning an error when we detect that the construction is wrong wouldn't be better. Such overwriting might be confusing and hard to debug and I cannot think of a single use case where we'd actually need it.
Ideally we would disallow it at compile time but that's not really an option here.

Reply:
Hmmm, I will revert it for now, and we can have a discussion later (now we are validating opts, we could validate builder constructs instead, so there is a room for discussion there).

Reply:
Done.